### PR TITLE
Misc trivial fixes and cleanups

### PR DESCRIPTION
**Pull request by scop**

### CODE REVIEW

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/CreateRequest.java [hunk 1]
```diff
@@ -216,7 +216,7 @@ private CreateRequest(ContentFormat format, LwM2mPath target, Integer instanceId
             if (instanceId == null) {
                 instanceId = target.getObjectInstanceId();
             } else {
-                if (instanceId != target.getObjectInstanceId()) {
+                if (instanceId.equals(target.getObjectInstanceId())) {
```
Comment (sbernard31) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/CreateRequest.java]: you meant !equals ?

File: leshan-core/src/main/java/org/eclipse/leshan/json/JsonArrayEntry.java [hunk 2]
```diff
@@ -155,7 +157,7 @@ public boolean equals(Object obj) {
                 return false;
         } else if (!stringValue.equals(other.stringValue))
             return false;
-        if (time != other.time)
```
Comment (sbernard31) [inline leshan-core/src/main/java/org/eclipse/leshan/json/JsonArrayEntry.java]: There is an mistake here, this method was modified manually. We should regenerated with eclipse : 
```java
if (time == null) {
    if (other.time != null)
        return false;
} else if (!time.equals(other.time))
    return false;
```

File: leshan-core/src/main/java/org/eclipse/leshan/json/JsonArrayEntry.java [hunk 3]
```diff
@@ -132,30 +132,15 @@ public boolean equals(Object obj) {
         if (getClass() != obj.getClass())
             return false;
```
Comment (sbernard31) [inline leshan-core/src/main/java/org/eclipse/leshan/json/JsonArrayEntry.java]: This equals methods is generated code. I'm not sure we want to modify/simplify it ?

File: leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java [hunk 4]
```diff
@@ -185,8 +185,6 @@ public static void createAndStartServer(int webPort, String localAddress, int lo
         }
 
         // Get public and private server key
-        PrivateKey privateKey = null;
-        PublicKey publicKey = null;
```
Comment (sbernard31) [inline leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java]: `publicKey` is used below out of the try/catch block : compilation error.

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/CreateRequest.java (hunk 1 above)
Comment (scop) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/CreateRequest.java]: Ah, yes.

File: leshan-core/src/main/java/org/eclipse/leshan/json/JsonArrayEntry.java (hunk 3 above)
Comment (scop) [inline leshan-core/src/main/java/org/eclipse/leshan/json/JsonArrayEntry.java]: Will remove this change.

File: leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java (hunk 4 above)
Comment (scop) [inline leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java]: Oops, sorry about that.

**Comment by scop** [comment]
Thanks for the comments, all should be addressed now.

**Comment by sbernard31** [comment]
Integrated in master. Thx @scop

**Pull request merged**